src/docfx/build/xref/XrefMap.cs

```diff
@@ -154,22 +151,45 @@ private static XrefSpec LoadSchemaDocument(List<Error> errors, JToken token, Doc
                     throw Errors.SchemaNotFound(file.Mime).ToException();
                 }
 
+                var uidExists = schema.Type.GetProperty("Uid") != null;
                 var (schemaErrors, content) = JsonUtility.ToObject(token, schema.Type, transform: TransformContent);
                 errors.AddRange(schemaErrors);
-                return xref;
+                if (!string.IsNullOrEmpty((string)schema.Type.GetProperty("Uid")?.GetValue(content, null)))
+                {
+                    return BuildXref();
+                }
+                else if (extensionData.Count > 0)
+                {
+                    errors.Add(Errors.UidMissing());
+                }
+                return null;
 
-                object TransformContent(DataTypeAttribute attribute, object value)
+                XrefSpec BuildXref()
                 {
-                    if (attribute is UidAttribute)
+                    var xref = new XrefSpec
                     {
-                        if (!string.IsNullOrEmpty((string)value))
-                        {
-                            // TODO: set name and extension data based on defined xref properties in the schema type
-                            xref = new XrefSpec { Uid = (string)value, Href = file.SitePath };
-                        }
+                        Uid = (string)schema.Type.GetProperty("Uid")?.GetValue(content, null),
+                        Name = (string)schema.Type.GetProperty("Name")?.GetValue(content, null),
+                        Href = file.SitePath,
+                    };
+                    xref.ExtensionData.Merge(extensionData);
+                    return xref;
+                }
+
+                object TransformContent(DataTypeAttribute attribute, object value, string jsonPath, string fieldName)
+                {
+                    if (attribute is XrefPropertyAttribute && IsRootLevel(jsonPath))
+                    {
+                        extensionData[fieldName] = (string)value;
                     }
                     return (string)value;
                 }
+
+                // e.g.: name => 1, data.name => 2
+                bool IsRootLevel(string path)
```

Comment:
this is wrong, a JSON field name can contain `.` like `ms.date`

Reply:
removed

---
In reply to: [220428046](https://github.com/dotnet/docfx/pull/3443#discussion_r220428046) [](ancestors = 220428046)